pkg/integration/helper.go

```diff
@@ -22,3 +23,46 @@ func ValidateURL(location string) error {
 	}
 	return nil
 }
+
+// ErrInvalidJSON is an error returned when a string is not a valid JSON.
+var ErrInvalidJSON = errors.New("invalid JSON")
+
+// ValidateJSON checks if the given string is a valid JSON and that there
+// is no space character in any of the keys and values of a JSON object.
+func ValidateJSON(p string, s string) error {
+	var data interface{}
+
+	if err := json.Unmarshal([]byte(s), &data); err != nil {
+		return fmt.Errorf("%w in --%s: %s", ErrInvalidJSON, p, err)
+	}
+
+	return checkJSONSpaces(p, data)
+}
+
+// ErrJSONContainsSpaces is an error returned when a JSON key or value contains spaces.
+var ErrJSONContainsSpaces = errors.New("contains unexpected spaces")
+
+func checkJSONSpaces(p string, data interface{}) error {
+	switch v := data.(type) {
+	case string:
+		if strings.Contains(v, " ") {
+			return fmt.Errorf("--%s %w in value %q", p, ErrJSONContainsSpaces, v)
+		}
+	case map[string]interface{}:
+		for key, val := range v {
+			if strings.Contains(key, " ") {
```

Comment:
You could potentially remove this check if you just call checkJSONSpaces on both key and value the check will be done on the next call.

Reply:
Done